src/webjure/json_schema/validator.clj

```diff
@@ -109,35 +128,34 @@
   (when enum
     (let [allowed-values (into #{} enum)]
       (when-not (allowed-values data)
-        {:error :invalid-enum-value
-         :data data
+        {:error          :invalid-enum-value
+         :data           data
          :allowed-values allowed-values}))))
-  
+
 (defmethod validate-by-type "string"
-  [_ data _]
-  (when-not (string? data)
-    {:error :wrong-type :expected :string :data data}))
+  [schema data _]
+  (if (not (string? data))
+    {:error :wrong-type :expected :string :data data}
+    (let [format (get schema "format")]
+      (when (and format (= format "date-time") (nil? (time/from-string data)))
+        {:error :wrong-type :expected :date :data data}))))
```

Comment:
Type is still string. Use :wrong-format or some other keyword to signal string format errors.